sway/tree/container.c

```diff
@@ -1241,6 +1236,50 @@ void container_floating_move_to(struct sway_container *con,
 	}
 }
 
+void container_floating_move_to_center(struct sway_container *con) {
+	if (!sway_assert(container_is_floating(con),
+			"Expected a floating container")) {
+		return;
+	}
+	struct sway_container *ws = container_parent(con, C_WORKSPACE);
+	double new_lx = ws->x + (ws->width - con->width) / 2;
+	double new_ly = ws->y + (ws->height - con->height) / 2;
+	container_floating_translate(con, new_lx - con->x, new_ly - con->y);
+}
+
+void container_floating_move_to_container(struct sway_container *container,
+		struct sway_container *destination) {
+	// Resolve destination into a workspace
+	struct sway_container *new_ws = NULL;
+	if (destination->type == C_OUTPUT) {
+		new_ws = output_get_active_workspace(destination->sway_output);
+	} else if (destination->type == C_WORKSPACE) {
+		new_ws = destination;
+	} else {
+		new_ws = container_parent(destination, C_WORKSPACE);
+	}
+	if (!new_ws) {
+		// This can happen if the user has run "move container to mark foo",
+		// where mark foo is on a hidden scratchpad container.
+		return;
+	}
+	struct sway_container *old_ws = container_parent(container, C_WORKSPACE);
+	if (old_ws != new_ws) {
+		container_remove_child(container);
+		container_add_child(new_ws->sway_workspace->floating, container);
```

Comment:
Needs a reparent signal event here so that the textures get updated when moved to an output with a different scale